### PR TITLE
SameSite support for cookies

### PR DESCRIPTION
**Pull request by danielhedren**
With the new secure-by-default approach chrome has taken, we need to be able to set the samesite attribute of cookies in some applications. The default behaviour is left as it was previously, i.e. samesite unset.

### CODE REVIEW

File: starlette/responses.py [hunk 1]
```diff
@@ -14,6 +14,11 @@
 from starlette.datastructures import URL, MutableHeaders
 from starlette.types import Receive, Scope, Send
 
+# Workaround for adding samesite support to pre 3.8 python
+from http.cookies import Morsel
+
+Morsel._reserved[str("samesite")] = str("SameSite")  # type: ignore
```
Comment (lovelydinosaur) [inline starlette/responses.py]: We shouldn't need the `str()` wrappers here.

(Eg. the comments against https://stackoverflow.com/questions/50813091/how-do-i-set-the-samesite-attribute-of-http-cookies-in-python?noredirect=1#comment93090548_50813092 indicate that they were only there for Python 2 support)

Comment (danielhedren) [reply starlette/responses.py]: Yeah, you're right.

**Comment by molten-firescar96** [comment]
Glad to see someone already made a PR. What else is needed before this is merged in?

**Comment by lovelydinosaur** [comment]
Okay, this is nice, aplogies for the slow uptake on it.

The `None` vs `"none"` case is a bit confusing. Given that browsers are [starting to default to "lax"](https://www.chromestatus.com/feature/5088147346030592) for the unset case ([MDN docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies#SameSite_cookies)), I wonder if we should:

* Switch to `"lax"` by default.
* Drop the `None` case - enforce that we always set something, that may be either `"strict"` `"lax"`, or `"none"`.

**Comment by danielhedren** [comment]
> 
> 
> Okay, this is nice, aplogies for the slow uptake on it.
> 
> The `None` vs `"none"` case is a bit confusing. Given that browsers are [starting to default to "lax"](https://www.chromestatus.com/feature/5088147346030592) for the unset case ([MDN docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies#SameSite_cookies)), I wonder if we should:
> 
>     * Switch to `"lax"` by default.
> 
>     * Drop the `None` case - enforce that we always set something, that may be either `"strict"` `"lax"`, or `"none"`.

I think defaulting to "lax" and dropping None is a good idea for the reasons you point out.

**Comment by lovelydinosaur** [comment]
Fantastic stuff, thanks!

File: docs/responses.md [hunk 2]
```diff
@@ -31,7 +31,7 @@ async def app(scope, receive, send):
 
 Starlette provides a `set_cookie` method to allow you to set cookies on the response object.
 
-Signature: `Response.set_cookie(key, value, max_age=None, expires=None, path="/", domain=None, secure=False, httponly=False)`
+Signature: `Response.set_cookie(key, value, max_age=None, expires=None, path="/", domain=None, secure=False, httponly=False, samesite=None)`
```
Comment (lovelydinosaur) [inline docs/responses.md]: Needs `samesite="lax"` here, too.

**Comment by lovelydinosaur** [comment]
Ace, good work!

**Comment by zero-shubham** [comment]
I don't see SameSite is being added when using response.set_cookie(). tried and tested with -> samesite=None samesite="none" samesite="lax" it should show up in headers when checking inside network tab right?

**Comment by lovelydinosaur** [comment]
You'd need to be testing against master - it's not in the latest release yet.

**Comment by zero-shubham** [comment]
> You'd need to be testing against master - it's not in the latest release yet.

Got it, for now this is my solution -> https://github.com/tiangolo/fastapi/issues/1099

**Pull request merged**